stdlib/event/time.lua

```diff
@@ -0,0 +1,73 @@
+--- Time Event module
+-- @module Event.Time
+
+require 'stdlib/event/event'
+
+Event.Time = {}
+Event.Time._last_change = {}
+
+--All times are offset by 0.5
+--This is because both EvoGUI and MoWeather already apply that offset.
+--Following the precedent to remain consistent.
+--Actually, this little snippet is even borrowed from EvoGUI.
+if remote.interfaces.MoWeather then
+    -- assume MoWeather's getdaytime is sane
+    function Event.Time.get_day_time(surface_name_or_index) return remote.call("MoWeather", "getdaytime", surface_name_or_index) end
+else
+    -- 0.5 is midnight; let's make days *start* at midnight instead.
+    function Event.Time.get_day_time(surface_name_or_index) return game.surfaces[surface_name_or_index].daytime + 0.5 end
+end
+
+--These are Surface only events
+--- @field Gets around ldoc V1.4.3 eating the first field. :(
+Event.Time.ldoc_V1_4_3_eats_this   = nil
```

Comment:
....And I finally figured out why it was getting eaten. Wonder if that is an ldoc bug?

```lua
--This comment makes the below @field get eatten
---@field The above comment ate me.
answer_to_life_the_universe_and_everything = 42